WalletWasabi.Tests/UnitTests/WabiSabi/JsonConverters/CompactSignatureJsonConverterTests.cs

```diff
@@ -23,10 +23,39 @@ public void SignatureTest()
 		Assert.True(privateKey.IsCompressed);
 
 		uint256 hashMsg = Hashes.DoubleSHA256(Encoding.ASCII.GetBytes("compact hashing test"));
-		byte[] compactSignature = privateKey.SignCompact(hashMsg);
+		CompactSignature compactSignature = privateKey.SignCompact(hashMsg);
 		Assert.NotNull(compactSignature);
 
-		string hex = ByteHelpers.ToHex(compactSignature);
+		byte[] sigBytes = CompactSignatureJsonConverter.ToBytes(compactSignature);
+		string hex = ByteHelpers.ToHex(sigBytes);
 		Assert.Equal("1F71932FFF735FA6A57787191A296717F71270B2B7E1D90008B7147117F250DBDE012359EED51D28682B1AAB686A8FD8A411A8D07F1EB4D7CDAC5B7EBE73F260A0", hex);
 	}
+
+	[Fact]
+	public void ConverterTest()
```

Comment:
This test is okay but my proposal went more in the direction of guaranteeing compatibility, i mean, encode a signature using the `ByteArrayJsonConverter` and the new `CompactSignatureJsonConverter` and verify both are identical. 

I cannot see that in this test without looking into the implementations and see that both encode the signature as base64 string.

Reply:
So I'm not sure I follow here but what I did is:

1. First commit 853c115f745819091bf1edcc30c9af43398cd8d0 shows a serialization [result](https://github.com/zkSNACKs/WalletWasabi/pull/7735/commits/853c115f745819091bf1edcc30c9af43398cd8d0#diff-c400eec6e3f6c8dedc392ab88ff4cceeb40cce75863205cd5df38fc63796b277R30) using NBitcoin 6 version
2. Second commit c400eec6e3f6c8dedc392ab88ff4cceeb40cce75863205cd5df38fc63796b277R26 shows that the serialization [did not change](https://github.com/zkSNACKs/WalletWasabi/pull/7735/commits/7954c00ea6222f411c29ec66530b7aeb359c3753#diff-c400eec6e3f6c8dedc392ab88ff4cceeb40cce75863205cd5df38fc63796b277R31) in NBitcoin 7.

Now given that NBitcoin 7 changes API, I'm not sure how to put old results and new results in a single commit. I might misunderstand your comment though.